api/app/Http/Controllers/DashboardController.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Announcement;
+use App\Models\PermissionRole;
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Auth;
+
+class DashboardController extends Controller
+{
+    public function index() 
+    {
+        $announcements = Announcement::all();
+
+        return response()->json($announcements);
+    }
+
+    public function store(Request $request)
+    {
+        $user = Auth::user();
+        
+        $user_fullname = $user['fullname'];
+        $role_users = $user->roleUsers;
+
+        $validatedData = $request->validate([
+            'announcementable_id' => 'required',
+            'announcementable_type' => 'required',
+            'announcement' => 'required'
+        ]);
+
+        //checks if role_users has permission canCreateAnnouncement 
+        //boolean values are set for each role the user has
+        //returns object
+        $role_permissions = $role_users->map(function ($item, $key) {
+            return PermissionRole::where('role_id', $item->role_id)
+            ->whereHas('permission', function ($query) {
+                $query->where('permission', 'canCreateAnnouncement');
+            })
+            ->exists(); 
+        });
+
+        //converts role_permissions object to array
+        $canCreateAnnouncementBoolean = in_array('true', $role_permissions->toArray());
```

Comment:
This line can also be eliminated if we use the role_id directly since ```$role_permissions``` will now return either 0 or 1